File: frontend/lib/core/models/saved.dart
import 'business.dart';

class SavedList {
  const SavedList({
    required this.id,
    required this.name,
    required this.tone,
    required this.count,
    required this.businesses,
  });

  final int id;
  final String name;
  final String tone;
  final int count;
  final List<BusinessCard> businesses;

  factory SavedList.fromJson(Map<String, dynamic> j) => SavedList(
        id: j['id'] as int,
        name: j['name'] as String,
        tone: j['tone'] as String? ?? 'gold',
        count: j['count'] as int? ?? 0,
        businesses: (j['businesses'] as List?)
                ?.map((e) => BusinessCard.fromJson(e as Map<String, dynamic>))
                .toList() ??
            const [],
      );
}

File: backend/app/schemas/saved.py
from pydantic import BaseModel, ConfigDict, Field

from app.schemas.business import BusinessCard


class SavedListCreate(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    tone: str = "gold"


class SavedListOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    tone: str
    count: int = 0
    businesses: list[BusinessCard] = Field(default_factory=list)


class SaveToListRequest(BaseModel):
    list_id: int | None = None  # when None, save to (or create) the default list
